pkgs/tools/inputmethods/fcitx5/fcitx5-rime-with-nix-env-variable.patch

```diff
@@ -0,0 +1,18 @@
+:100644 100644 fac4f53 aed9617 M	src/rimeengine.cpp
+
+diff --git a/src/rimeengine.cpp b/src/rimeengine.cpp
+index fac4f53..aed9617 100644
+--- a/src/rimeengine.cpp
++++ b/src/rimeengine.cpp
+@@ -164,7 +164,10 @@ void RimeEngine::rimeStart(bool fullcheck) {
+             RIME_ERROR() << "Failed to create user directory: " << userDir;
+         }
+     }
+-    const char *sharedDataDir = RIME_DATA_DIR;
++    const char *sharedDataDir = getenv("NIX_RIME_DATA_DIR");
++    if (!sharedDataDir) {
```

Comment:
Can this only support one directory? It would be easier if we could append it instead.

Reply:
It only supports a single directory. We will have to modify librime if we want support for multiple directories.

Reply:
The librime only supports for one dictory during built time. This patch is make that dictory working in runtime for fcitx5-rime. If we want suuport for multiple directories, we have to patch librime instead of this.